obsproject/obs-deps
# Build windows deps

### Description


This adds scripts to cross-compile the Windows dependencies from Ubuntu.  Starting from Jim's build scripts, this gets us to a point where we can automate building the FFmpeg related dependencies on the obs-deps CI.

Obsoletes #1.

Some known issues:
* ~~The current deps package contains `zlib.dll` and `zlib.lib`.  This script generates `libzlib.dll` and `libzlib.lib`.  I couldn't figure out why this happens, so I assume Jim's scripts have a modification that I don't have here.~~
* This appears to _only_ be the FFmpeg related dependencies.  It does not build other dependencies such as cURL, FreeType, Lua/LuaJIT, Python, RNNoise, SWIG, Vulkan, or Websockets.  I'm guessing Jim builds those elsewhere.
* This doesn't follow the paradigms with GitHub workflows and script generation used by the macOS scripts.  I worked on this before that was merged, and I'd rather get this off my computer so it stops being neglected there.  If need be, it can be reworked later to fit those paradigms.

#### To-do
- [x] Fix naming of zlib DLL from libzlib.dll to zlib.dll as the current deps packages have
- [x] Moves files in `patches` to `patch` to avoid duplicate patch directories
- [ ] Address extra files produced here compared to current deps packages
- [ ] Build other deps not included here (perhaps in a follow-up PR rather than here)

### Motivation and Context



Be more transparent about our dependencies.  Automate the process of updating and deploying our dependencies.

### How Has This Been Tested?



I used the deps built by these scripts as a replacement for the current Windows deps and they seemed to work fine.  Streaming and recording worked.  Admittedly, I did this testing a few months ago, so we should probably retest more thoroughly.

To test these:
1. Download and extract the deps artifact
2. Copy the artifact's `bin` and `include` folders over the existing corresponding deps (win32 or win64)
3. Clear the OBS build folder
4. Reset the additional_install_files folder (delete the folder and then `git restore` or `git reset`)
5. Run CMake for OBS as usual to configure and generate the project
6. Open the project and build it
7. Run OBS and record, stream, or perform other actions to test


### Types of changes


- New feature (non-breaking change which adds functionality)





### Checklist:
